control-plane-operator/hostedclusterconfigoperator/controllers/node/node.go

```diff
@@ -0,0 +1,95 @@
+package node
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	supportutil "github.com/openshift/hypershift/support/util"
+
+	"github.com/openshift/hypershift/support/upsert"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	nodePoolAnnotation = "hypershift.openshift.io/nodePool"
```

Comment:
s/nodePoolAnnotation/nodePoolLabel/

Reply:
should this be part of the API?

Reply:
It's both annotation and label. I updated to move the label declaration into API.